OneCommunityGlobal/HGNRest
# Jiayu Sort projects by most recent edited inventories and members

# Description
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/58479595/cc5552db-bacf-486e-aa6d-22ffad0eb4f9)

This is only for sorting the **members** since inventory is not in use.

## Related PRS (if any):
To test this backend PR you need to checkout [frontend PR2443](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2443).

## Main changes explained:
- Update `projectController.js` to include `membersModifiedDatetime` field when fetching the projects, and set the `membersModifiedDatetime` field to the current time when assigning/removing members from project.
- Update `userProfileController` to include the logic to check which projects have changed when the user profile is updated, and update the `membersModifiedDatetime` field for the projects changed.
- Update `src/models/project.js` to update the project model to include `membersModifiedDatetime` field.
- Update `userProfileRouter.js` to include the project model in the userprofile controller
- Update `routes.js` to include project model in the userProfile router.

## How to test:
1. check into current branch
2. do `npm install` and `npm run build && npm run start` to run this PR locally
3. Please follow the test steps in the frontend PR to test. 

## Screenshots or videos of changes:
https://www.loom.com/share/f90a2754cee444f8ae711a0a0f09c749?sid=3a343985-0bc6-480a-9942-b18bda25ec1e
